infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/VcsSshKeysProvisioner.java

```diff
@@ -0,0 +1,218 @@
+                .withNewMetadata()
+                .withName("sshconfigmap")
+                .endMetadata()
+                .withData(map)
+                .build();
+
+        k8sEnv.getConfigMaps().put(configMap.getMetadata().getName(), configMap);
+        k8sEnv.getPodsData().values().forEach(p -> addConfigFile(p.getSpec()));
+      }
+    } catch (ServerException e) {
+      LOG.warn("Unable get SSH Keys ", e);
+    }
+  }
+
+  private void addSshKeySecret(String secretName, PodSpec podSpec) {
+    podSpec
+        .getVolumes()
+        .add(
+            new VolumeBuilder()
+                .withName(secretName)
+                .withSecret(new SecretVolumeSourceBuilder().withSecretName(secretName).build())
+                .build());
+    List<Container> containers = podSpec.getContainers();
+    containers.forEach(
+        container -> {
+          VolumeMount volumeMount =
+              new VolumeMountBuilder()
+                  .withName(secretName)
+                  .withNewReadOnly(false)
+                  .withReadOnly(false)
+                  .withMountPath(SSH_BASE_CONFIG_PATH + secretName)
+                  .build();
+          container.getVolumeMounts().add(volumeMount);
+        });
+  }
+
+  private void addConfigFile(PodSpec podSpec) {
+    podSpec
+        .getVolumes()
+        .add(
+            new VolumeBuilder()
+                .withName("configvolume")
+                .withConfigMap(new ConfigMapVolumeSourceBuilder().withName("sshconfigmap").build())
+                .build());
+
+    List<Container> containers = podSpec.getContainers();
+    containers.forEach(
+        container -> {
+          VolumeMount volumeMount =
+              new VolumeMountBuilder()
+                  .withName("configvolume")
+                  .withMountPath(SSH_CONFIG_PATH)
+                  .withSubPath(SSH_CONFIG)
+                  .withReadOnly(false)
+                  .withNewReadOnly(false)
+                  .build();
+          container.getVolumeMounts().add(volumeMount);
+        });
+  }
+
+  /**
+   * @param host the host of version control service (e.g. github.com, gitlab.com and etc)
+   * @return the ssh configuration which include host of host and identity file location
+   *     <p>Example of provided configuration:
+   *     <p>host github.com HostName github.com IdentityFile /etc/ssh/github-com/ssh-privatekey
+   */
+  private String buildConfig(@NotNull String host) {
+    String validName = getValidNameForSecret(host);
+    StringBuilder stringBuilder = new StringBuilder();
+    stringBuilder
+        .append("host ")
+        .append(host)
+        .append("\n")
+        .append("HostName ")
+        .append(host)
+        .append("\nIdentityFile ")
+        .append(SSH_BASE_CONFIG_PATH)
+        .append(validName)
+        .append("/")
+        .append(SSH_PRIVATE_KEY)
+        .append("\n");
+    return stringBuilder.toString();
+  }
+
+  /**
+   * Check is given name available for secret name
```

Comment:
I think java doc is a bit misleading. In my understanding `Check` should return true/false or throw an exception in case of invalid value. But this method transforms any value to a valid one.

Reply:
fixed